ext/pdo/pdo_stmt.c

```diff
@@ -1209,9 +1228,8 @@ PHP_METHOD(PDOStatement, fetchObject)
 	PHP_STMT_GET_OBJ;
 	PDO_STMT_CLEAR_ERR();
 
-	if (!pdo_stmt_verify_mode(stmt, how, 0)) {
-		RETURN_FALSE;
-	}
+	/* use pdo_stmt_verify_mode() to set fetch mode for this specific statement */
+	ZEND_ASSERT(pdo_stmt_verify_mode(stmt, PDO_FETCH_CLASS, 0, false));
```

Comment:
You need to move the call outside the assert, otherwise it's not guaranteed to be executed.